addons/edi/models/ir_attachment.py

```diff
@@ -0,0 +1,38 @@
+from odoo import api, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+
+import fnmatch
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    @api.constrains("res_model")
+    def check_filename(self):
```

Comment:
Q: Do we need this? Can't see it being called anywhere...

Reply:
This function is called when res_model field of the ir.attachment model changes - when a record is attached to a document. There is an enforce_filename field in the edi.document model which is set for each document type, I ported this over as it is set as True for the Item master document.

I think "enforce_filename" is a UDES field, I am now thinking it should be "u_enforce_filename"

Reply:
Sorry this whole comment thread was a disaster on my part. We are in the `odoo-edi` repo, so this is the repo that adds `edi.document`, and as it is a new model we don't need any field prefixes.

Sorry again! Resolving now...